extensions/ql-vscode/src/local-databases-ui.ts

```diff
@@ -754,31 +776,40 @@ export class DatabaseUI extends DisposableObject {
    */
   private async chooseAndSetDatabase(
     byFolder: boolean,
-    progress: ProgressCallback,
-    token: CancellationToken,
+    progress: ProgressContext | undefined,
   ): Promise<DatabaseItem | undefined> {
     const uri = await chooseDatabaseDir(byFolder);
     if (!uri) {
       return undefined;
     }
 
-    if (byFolder) {
-      const fixedUri = await this.fixDbUri(uri);
-      // we are selecting a database folder
-      return await this.setCurrentDatabase(progress, token, fixedUri);
-    } else {
-      // we are selecting a database archive. Must unzip into a workspace-controlled area
-      // before importing.
-      return await importArchiveDatabase(
-        this.app.commands,
-        uri.toString(true),
-        this.databaseManager,
-        this.storagePath,
-        progress,
-        token,
-        this.queryServer?.cliServer,
-      );
-    }
+    return await withInheritedProgress(
+      progress,
+      async (progress, token) => {
```

Comment:
I see that you're not inheriting and cancellation token.  I haven't looked closely, but I would have expected this to be the case.

Reply:
The `ProgressContext` contains both the progress callback and the token from the parent. If there's a parent context, we'll inherit its token. Otherwise, `withProgress` will create a new one if needed.